### PR TITLE
#96 Docker with psql

### LINKED ISSUE #96: Настроить PostgreSQL в проекте
(Docker + запуск локально проекта) и убеждается, что работает CRUD.

### PR DESCRIPTION
**Pull request by zemer**
`=> ERROR [4/4] RUN npm install && npm run build                                                                                                                      44.1s 
------
 > [4/4] RUN npm install && npm run build:
#6 15.30 npm WARN connected-react-router@6.8.0 requires a peer of immutable@^3.8.1 || ^4.0.0-rc.1 but none is installed. You must install peer dependencies yourself.       
#6 15.30 npm WARN connected-react-router@6.8.0 requires a peer of seamless-immutable@^7.1.3 but none is installed. You must install peer dependencies yourself.
#6 15.30 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules/fsevents):
#6 15.30 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})  
#6 15.30 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules/webpack-dev-server/node_modules/fsevents):
#6 15.30 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"}) 
#6 15.30
#6 15.30 audited 1983 packages in 14.476s
#6 17.02
#6 17.02 149 packages are looking for funding
#6 17.02   run `npm fund` for details
#6 17.02
#6 17.02 found 0 vulnerabilities
#6 17.02
#6 17.49
#6 17.49 > reactive-battle@1.0.0 build /project
#6 17.49 > webpack --config ./webpack/webpack.client.prod.js
#6 17.49
#6 43.60 assets by status 1.5 MiB [cached] 13 assets
#6 43.60 Entrypoint main = vendors.8e81299b09fe3f00310d.js main.2b62892ef1244bd31b8a.css main.e34ca03502b7e4660dd0.js 5 auxiliary assets
#6 43.60 orphan modules 76.5 KiB [orphan] 35 modules
#6 43.60 runtime modules 4.02 KiB 8 modules
#6 43.60 modules by path ./src/ 170 KiB (javascript) 64.7 KiB (css/mini-extract)
#6 43.60   cacheable modules 170 KiB 132 modules
#6 43.60   css modules 64.7 KiB
#6 43.60     modules by path ./src/pages/ 2.94 KiB 12 modules
#6 43.60     modules by path ./src/components/ 3.65 KiB 10 modules
#6 43.60     modules by path ./src/styles/ 58.1 KiB 2 modules
#6 43.60 modules by path ./node_modules/ 357 KiB 17 modules
#6 43.60
#6 43.60 LOG from InjectManifest
#6 43.60 <i> The service worker at service-worker.js will precache
#6 43.60 <i>         13 URLs, totaling 1.55 MB.
#6 43.60
#6 43.60 ERROR in ./src/images/no-avatar.png
#6 43.60 Module Error (from ./node_modules/image-minimizer-webpack-plugin/dist/loader.js):
#6 43.60 spawn /project/node_modules/optipng-bin/vendor/optipng ENOENT
#6 43.60  @ ./src/components/avatar/Avatar.tsx 19:19-64
#6 43.60  @ ./src/components/avatar/index.ts 7:15-34
#6 43.60  @ ./src/pages/profileEdit/ProfileEdit.tsx 31:33-62
#6 43.60  @ ./src/components/app/App.tsx 13:38-84
#6 43.60  @ ./src/index.tsx 28:30-60
#6 43.60
#6 43.60 ERROR in ./src/images/sprite-base.png
#6 43.60 Module Error (from ./node_modules/image-minimizer-webpack-plugin/dist/loader.js):
#6 43.60 spawn /project/node_modules/optipng-bin/vendor/optipng ENOENT
#6 43.60  @ ./src/engine/SpritesSheet.ts 4:14-61
#6 43.60  @ ./src/engine/sprites/Sprite.ts 4:23-49
#6 43.60  @ ./src/engine/sprites/Tank.ts 10:33-66
#6 43.60  @ ./src/engine/sprites/enemies/EnemyTank.ts 7:31-62
#6 43.60  @ ./src/pages/game/Game.tsx 30:36-80
#6 43.60  @ ./src/components/app/App.tsx 11:31-63
#6 43.60  @ ./src/index.tsx 28:30-60
#6 43.60
#6 43.60 ERROR in ./src/pages/error5xx/error5xx.scss
#6 43.60 Module build failed (from ./node_modules/mini-css-extract-plugin/dist/loader.js):
#6 43.60 ModuleError: Module Error (from ./node_modules/image-minimizer-webpack-plugin/dist/loader.js):
#6 43.60 spawn /project/node_modules/optipng-bin/vendor/optipng ENOENT
#6 43.60     at Object.emitError (/project/node_modules/webpack/lib/NormalModule.js:448:6)
#6 43.60     at /project/node_modules/image-minimizer-webpack-plugin/dist/loader.js:40:12
#6 43.60     at Array.forEach (<anonymous>)
#6 43.60     at Object.loader (/project/node_modules/image-minimizer-webpack-plugin/dist/loader.js:39:19)
#6 43.60  @ ./src/pages/error5xx/Error5xx.tsx 7:0-26
#6 43.60  @ ./src/pages/error5xx/index.ts 7:17-38
#6 43.60  @ ./src/components/app/App.tsx 16:35-66
#6 43.60  @ ./src/index.tsx 28:30-60
#6 43.60
#6 43.60 ERROR in ./src/pages/error404/error404.scss
#6 43.60 Module build failed (from ./node_modules/mini-css-extract-plugin/dist/loader.js):
#6 43.60 ModuleError: Module Error (from ./node_modules/image-minimizer-webpack-plugin/dist/loader.js):
#6 43.60 spawn /project/node_modules/jpegtran-bin/vendor/jpegtran ENOENT
#6 43.60     at Object.emitError (/project/node_modules/webpack/lib/NormalModule.js:448:6)
#6 43.60     at /project/node_modules/image-minimizer-webpack-plugin/dist/loader.js:40:12
#6 43.60     at Array.forEach (<anonymous>)
#6 43.60     at Object.loader (/project/node_modules/image-minimizer-webpack-plugin/dist/loader.js:39:19)
#6 43.60  @ ./src/pages/error404/Error404.tsx 7:0-26
#6 43.60  @ ./src/pages/error404/index.ts 7:17-38
#6 43.60  @ ./src/components/app/App.tsx 15:35-66
#6 43.60  @ ./src/index.tsx 28:30-60
#6 43.60
#6 43.60 2 ERRORS in child compilations
#6 43.60 webpack 5.20.2 compiled with 6 errors in 22951 ms
#6 43.68 npm ERR! code ELIFECYCLE
#6 43.68 npm ERR! errno 1
#6 43.68 npm ERR! reactive-battle@1.0.0 build: `webpack --config ./webpack/webpack.client.prod.js`
#6 43.68 npm ERR! Exit status 1
#6 43.68 npm ERR!
#6 43.68 npm ERR! Failed at the reactive-battle@1.0.0 build script.
#6 43.68 npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
#6 43.70
#6 43.70 npm ERR! A complete log of this run can be found in:
#6 43.70 npm ERR!     /root/.npm/_logs/2021-02-16T09_44_34_343Z-debug.log
------
executor failed running [/bin/sh -c npm install && npm run build]: exit code: 1`

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,73 @@
+FROM node:13
```
Comment (alexch365) [inline Dockerfile]: возможно ошибки из-за старой ноды, попробуй node:15-buster
https://hub.docker.com/_/node/

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    environment:
+      DATABASE_URL: postgres://postgres:newPassword@postgres:5432/my-db-name
+      NODE_ENV: development
+      PORT: 5000
+    ports:
+      - "6000:5000"
+    restart: always
+  postgres:
+    container_name: psql
+    image: postgres:12
```
Comment (alexch365) [inline docker-compose.yml]: можно сразу 13-й поставить

**Review by alexch365 — CHANGES_REQUESTED** [review]

**Comment by zemer** [comment]
В целом работает.

**Review by shchegol — APPROVED** [review]

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  postgres:
+    container_name: psql
+    image: postgres:13
+    ports:
+      - "5432:5432"
+    volumes:
+      - ./postgresdata:/var/lib/postgresql/data
+    restart: always
+    environment:
+      POSTGRES_USER: postgres
```
Comment (alexch365) [inline docker-compose.yml:24]: надо бы в отдельный файлик вынести юзера, пароль и название базы
и загитигнорить его

**Review by alexch365 — APPROVED** [review]

**Pull request merged**